src/main/java/com/denizenscript/depenizen/bukkit/events/crackshot/crackshotLandmineTriggerEvent.java

```diff
@@ -0,0 +1,80 @@
+package com.denizenscript.depenizen.bukkit.events.crackshot;
+
+import com.denizenscript.denizen.events.BukkitScriptEvent;
+import com.denizenscript.denizen.objects.EntityTag;
+import com.denizenscript.denizen.objects.PlayerTag;
+import com.denizenscript.denizen.utilities.implementation.BukkitScriptEntryData;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.scripts.ScriptEntryData;
+import com.shampaggon.crackshot.events.WeaponTriggerEvent;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+
+public class crackshotLandmineTriggerEvent extends BukkitScriptEvent implements Listener {
+
+    // <--[event]
+    // @Events
+    // crackshot landmine triggered
+    //
+    // @Regex ^on crackshot landmine triggered$
+    //
+    // @Triggers when a CrackShot landmine is triggered by an entity walking through it.
+    //
+    // @Cancellable true
+    //
+    // @Context
+    // <context.weapon> returns the weapon name of the landmine.
+    // <context.victim> returns the EntityTag of the entity that triggered the landmine.
+    //
+    // @Plugin Depenizen, CrackShot
+    //
+    // @Player Always
+    //
+    // -->
+
+    public static crackshotLandmineTriggerEvent instance;
+    public        WeaponTriggerEvent            event;
+    public        PlayerTag                     player;
+    public        ElementTag                    weaponName;
+    public        EntityTag                     victim;
+
+    public crackshotLandmineTriggerEvent() {
+        instance = this;
+    }
+
+    @Override
+    public ScriptEntryData getScriptEntryData() {
+        return new BukkitScriptEntryData(player, null);
+    }
+
+    @Override
+    public boolean couldMatch(ScriptPath path) {
+        return path.eventLower.startsWith("crackshot landmine triggered");
+    }
+
+    @Override
+    public String getName() {
+        return "CrackShotLandmineTriggereEvent";
+    }
+
+    @Override
+    public ObjectTag getContext(String name) {
+        if (name.equals("victim")) {
+            return victim;
+        }
+        else if (name.equals("weapon_name")) {
+            return weaponName;
+        }
+        return super.getContext(name);
+    }
+
+    @EventHandler
+    public void onCrackShotLandmineTriggerEvent(WeaponTriggerEvent event) {
+        player = PlayerTag.mirrorBukkitPlayer(event.getPlayer());
+        weaponName = new ElementTag(event.getWeaponTitle());
+        victim = new EntityTag(event.getVictim());
```

Comment:
None of these should be processed here. Use the `getContext` method

Reply:
I have moved all of these to the getContext method.